Compiler.Lexer/Compiler.Lexer/Tokenizer.cs

```diff
@@ -0,0 +1,403 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Compiler.Lexer
+{
+    public partial class Tokenizer
+    {
+        private CharStream Stream;
+        private Token LastToken;
+
+        public Tokenizer(string source)
+        {
+            Stream = new CharStream(source);
+            LastToken = new Token
+                (TokenKind.UnknownToken, string.Empty, 0, 0);
+        }
+
+        public Token Get()
+        {
+            Token get;
+            int lineNumber = Stream.CurrentLineNumber;
+            int position = Stream.CurrentPosition;
+
+            do
+            {
+                if (Stream.EOF)
+                {
+                    return null;
+                }
+            } while (IsWhiteSpace());
+
```

Comment:
Might be an option to condense following code to be able to read all calls to set "get" on the same page using null conditional operators. Less lines would be easier to scroll and easier to see more on the page at one time. Eliminates repetitive code for each call.
Possibly something like (but not positive)
return IsSymbol() ??
          IsOperator() ??
          .... etc....
         new Token(TokenKind.UnknownToken, Stream.Get().ToString(), lineNumber, position);

Reply:
If there is an idea about which calls find a match more often than others, order them in descending order by matches most often to matches least often. Will find the match faster for efficiency.